nml/actions/actionF.py

```diff
@@ -108,15 +108,19 @@ def prepare_output(self, sprite_num):
                 named_numbers[self.name.value] = self.id_number # Add name to the set 'safe' names.
         else: numbered_numbers.add(self.id_number) # Add number to the set of 'safe' numbers.
 
+        # Don't overlap with existing action F
+        if len(town_names_blocks) == 0: startbit = 0
+        else: startbit = max(block.free_bit for block in town_names_blocks.values())
+
         town_names_blocks[self.id_number] = self # Add self to the available blocks.
 
         # Ask descendants for the lowest available bit.
-        if len(blocks) == 0: startbit = 0 # No descendants, all bits are free.
-        else: startbit = max(town_names_blocks[block].free_bit for block in blocks)
+        if len(blocks) != 0: startbit = max(town_names_blocks[block].free_bit for block in blocks)
```

Comment:
They may again create overlapping bits.
Shouldn't it always use the result from line 112/113?